File: backend/app/models/student_status.py
from .. import db
from .student import Student

class StudentStatus(db.Model):
    __tablename__ = 'student_status'
    
    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    student_id = db.Column(db.String(20), db.ForeignKey('students.student_id'), nullable=False)
    name = db.Column(db.String(50), nullable=False)
    gender = db.Column(db.String(10), nullable=False)
    class_ = db.Column('class', db.String(20), nullable=False)
    grade = db.Column(db.String(20), nullable=False)
    contact = db.Column(db.String(100))
    status = db.Column(db.String(20), nullable=False, default='active')
    
    # 关联关系
    student = db.relationship('Student', backref=db.backref('status_info', lazy=True))
    
    def to_dict(self):
        # 状态文本映射
        status_map = {
            'active': '在校',
            'suspended': '休学',
            'graduated': '毕业',
            'dropped': '退学'
        }
        
        return {
            'id': self.student_id,
            'name': self.name,
            'gender': self.gender,
            'class': self.class_,
            'grade': self.grade,
            'contact': self.contact,
            'status': self.status,
            'statusText': status_map.get(self.status, '在校')
        }
    
    def __repr__(self):
        return f'<StudentStatus Student:{self.student_id} Status:{self.status}>'

File: frontend/src/stores/studentStatus.ts
import { defineStore } from 'pinia';
import type { StudentStatus } from '../types';

// 学籍存储
export const useStudentStatusStore = defineStore('studentStatus', {
  state: () => ({
    students: [] as StudentStatus[],
    loading: false,
    error: null as string | null,
  }),

  getters: {
    getStudentById: (state) => (id: string) => {
      return state.students.find(student => student.id === id);
    },

    getStudentsByGrade: (state) => (grade: string) => {
      return state.students.filter(student => student.grade === grade);
    },

    getStudentsByClass: (state) => (grade: string, className: string) => {
      return state.students.filter(student => student.grade === grade && student.class === className);
    },

    getStudentsByStatus: (state) => (status: string) => {
      return state.students.filter(student => student.status === status);
    },

    getActiveStudents: (state) => {
      return state.students.filter(student => student.status === 'active');
    },

    getSuspendedStudents: (state) => {
      return state.students.filter(student => student.status === 'suspended');
    },

    getGraduatedStudents: (state) => {
      return state.students.filter(student => student.status === 'graduated');
    },

    getDroppedStudents: (state) => {
      return state.students.filter(student => student.status === 'dropped');
    },

    getArchivedStudents: (state) => {
      return state.students.filter(student => student.status === 'graduated');
    },
  },

  actions: {
    // 初始化学籍数据
    initializeStudents(students: StudentStatus[]) {
      this.students = students;
    },

    // 添加学生
    addStudent(student: StudentStatus) {
      this.students.push(student);
    },

    // 更新学生
    updateStudent(updatedStudent: StudentStatus) {
      const index = this.students.findIndex(s => s.id === updatedStudent.id);
      if (index !== -1) {
        this.students[index] = updatedStudent;
      }
    },

    // 删除学生
    deleteStudent(id: string) {
      this.students = this.students.filter(student => student.id !== id);
    },

    // 更新学籍状态
    updateStudentStatus(studentId: string, status: 'active' | 'suspended' | 'graduated' | 'dropped') {
      const student = this.getStudentById(studentId);
      if (student) {
        student.status = status;
        student.statusText = this.getStatusText(status);
        this.updateStudent(student);
      }
    },

    // 归档学生（设置为毕业状态）
    archiveStudent(studentId: string) {
      this.updateStudentStatus(studentId, 'graduated');
    },

    // 取消归档学生（设置为在校状态）
    unarchiveStudent(studentId: string) {
      this.updateStudentStatus(studentId, 'active');
    },

    // 获取状态文本
    getStatusText(status: string): string {
      const statusMap: Record<string, string> = {
        'active': '在校',
        'suspended': '休学',
        'graduated': '毕业',
        'dropped': '退学'
      };
      return statusMap[status] || status;
    },

    // 清除所有学籍数据
    clearStudents() {
      this.students = [];
    },
  },
});
